service/kernel/data_management/train_test_split.cpp

```diff
@@ -0,0 +1,253 @@
+    DAAL_OVERFLOW_CHECK_BY_MULTIPLICATION(size_t, nTrainRows + nTestRows, nColumns);
+
+    NumericTableDictionaryPtr tableFeaturesDict = inputTable->getDictionarySharedPtr();
+
+    if (layout == NTLayout::soa)
+    {
+        BlockDescriptor<IdxType> trainIdxBlock, testIdxBlock;
+        trainIdxTable->getBlockOfColumnValues(0, 0, nTrainRows, readOnly, trainIdxBlock);
+        testIdxTable->getBlockOfColumnValues(0, 0, nTestRows, readOnly, testIdxBlock);
+        const IdxType * trainIdx = trainIdxBlock.getBlockPtr();
+        const IdxType * testIdx  = testIdxBlock.getBlockPtr();
+        DAAL_CHECK_MALLOC(trainIdx);
+        DAAL_CHECK_MALLOC(testIdx);
+
+        runBlocks(nColumns > 1 && nColumns * (nTrainRows + nTestRows) > THREADING_BORDER && nThreads > 1, nColumns, [&](size_t iCol) {
+            switch ((*tableFeaturesDict)[iCol].getIndexType())
+            {
+            case daal::data_management::features::IndexNumType::DAAL_FLOAT32:
+                s |= splitColumn<float, IdxType, cpu>(inputTable, trainTable, testTable, trainIdx, testIdx, nTrainRows, nTestRows, iCol, nThreads);
+                break;
+            case daal::data_management::features::IndexNumType::DAAL_FLOAT64:
+                s |= splitColumn<double, IdxType, cpu>(inputTable, trainTable, testTable, trainIdx, testIdx, nTrainRows, nTestRows, iCol, nThreads);
+                break;
+            default: s |= splitColumn<int, IdxType, cpu>(inputTable, trainTable, testTable, trainIdx, testIdx, nTrainRows, nTestRows, iCol, nThreads);
+            }
+        });
+    }
+    else
+    {
+        switch ((*tableFeaturesDict)[0].getIndexType())
+        {
+        case daal::data_management::features::IndexNumType::DAAL_FLOAT32:
+            s |= splitRows<float, IdxType, cpu>(inputTable, trainTable, testTable, trainIdxTable, testIdxTable, nTrainRows, nTestRows, nColumns,
+                                                nThreads);
+            break;
+        case daal::data_management::features::IndexNumType::DAAL_FLOAT64:
+            s |= splitRows<double, IdxType, cpu>(inputTable, trainTable, testTable, trainIdxTable, testIdxTable, nTrainRows, nTestRows, nColumns,
+                                                 nThreads);
+            break;
+        default:
+            s |= splitRows<int, IdxType, cpu>(inputTable, trainTable, testTable, trainIdxTable, testIdxTable, nTrainRows, nTestRows, nColumns,
+                                              nThreads);
+        }
+    }
+    return s;
+}
+
+template <typename IdxType>
+void trainTestSplitDispImpl(const NumericTablePtr inputTable, const NumericTablePtr trainTable, const NumericTablePtr testTable,
+                            const NumericTablePtr trainIdxTable, const NumericTablePtr testIdxTable)
+{
+#define DAAL_TTS(cpuId, ...) trainTestSplitImpl<IdxType, cpuId>(__VA_ARGS__);
+    DAAL_DISPATCH_FUNCTION_BY_CPU(DAAL_TTS, inputTable, trainTable, testTable, trainIdxTable, testIdxTable);
+#undef DAAL_TTS
```

Comment:
Why do we need this new define?
Even if it's needed - better to change the naming.

Reply:
Defines is needed to pass function in dispatcher. Name is changed to DAAL_TRAIN_TEST_SPLIT

Reply:
Ok, but I'm sure that you can do it without additional define.